controllers/front/ProductController.php

```diff
@@ -955,7 +955,7 @@ private function getIdProductAttributeByRequestOrGroup()
      */
     private function tryToGetAvailableIdProductAttribute($checkedIdProductAttribute)
     {
-        if (!Configuration::get('PS_DISP_UNAVAILABLE_ATTR')) {
+        if (!Product::isAvailableWhenOutOfStock($this->product->out_of_stock) && Configuration::get('PS_DISP_UNAVAILABLE_ATTR') == 0) {
```

Comment:
Hi @Shoprunners, do you think we can improve the readability of this function as it gets more complex now ? Either using a code comment or using a variable name*

*Example
```
$myVar = (!Product::isAvailableWhenOutOfStock($this->product->out_of_stock) && Configuration::get('PS_DISP_UNAVAILABLE_ATTR') == 0);
if ($myVar) {
```

Reply:
I've now added a comment similar to the one at the other site using the exact logic while also referring to that function.